src/main/java/org/gridsuite/study/server/service/ConsumerService.java

```diff
@@ -398,367 +238,287 @@ public Consumer<Message<String>> consumeCaseImportFailed() {
+                        updateType = NotificationService.UPDATE_TYPE_SECURITY_ANALYSIS_STATUS;
+                    } break;
+                    case SENSITIVITY_ANALYSIS: {
+                        networkModificationTreeService.updateSensitivityAnalysisResultUuid(receiverObj.getNodeUuid(), null);
+                        updateType = NotificationService.UPDATE_TYPE_SENSITIVITY_ANALYSIS_STATUS;
+                    } break;
+                    case SHORT_CIRCUIT: {
+                        // TODO : this Stopped function is the only one when SHORT_CIRCUIT doesn't handle buses => might cause a bug
+                        networkModificationTreeService.updateShortCircuitAnalysisResultUuid(receiverObj.getNodeUuid(), null);
+                        updateType = NotificationService.UPDATE_TYPE_SHORT_CIRCUIT_STATUS;
+                    } break;
+                    case VOLTAGE_INITIALIZATION: {
+                        networkModificationTreeService.updateVoltageInitResultUuid(receiverObj.getNodeUuid(), null);
+                        updateType = NotificationService.UPDATE_TYPE_VOLTAGE_INIT_STATUS;
+                    } break;
+                    case DYNAMIC_SIMULATION: {
+                        networkModificationTreeService.updateDynamicSimulationResultUuid(receiverObj.getNodeUuid(), null);
+                        updateType = NotificationService.UPDATE_TYPE_DYNAMIC_SIMULATION_STATUS;
+                    } break;
+                }
 
-                    // delete sensitivity analysis result in database
-                    updateSensitivityAnalysisResultUuid(receiverObj.getNodeUuid(), null);
+                UUID studyUuid = networkModificationTreeService.getStudyUuidForNodeId(receiverObj.getNodeUuid());
+                // send notification for stopped computation
+                notificationService.emitStudyChanged(studyUuid, receiverObj.getNodeUuid(), updateType);
+            } catch (JsonProcessingException e) {
+                LOGGER.error(e.toString());
+            }
+        }
+    }
 
-                    // send notification for stopped computation
-                    UUID studyUuid = networkModificationTreeService.getStudyUuidForNodeId(receiverObj.getNodeUuid());
-                    notificationService.emitStudyChanged(studyUuid, receiverObj.getNodeUuid(), NotificationService.UPDATE_TYPE_SENSITIVITY_ANALYSIS_STATUS);
-                } catch (JsonProcessingException e) {
-                    LOGGER.error(e.toString());
+    public void consumeCalculationResult(Message<String> msg, ComputationType computationType) {
+        UUID resultUuid = null;
+        String resultId = msg.getHeaders().get(RESULT_UUID, String.class);
+        if (resultId != null) {
+            resultUuid = UUID.fromString(resultId);
+        }
+        String receiver = msg.getHeaders().get(HEADER_RECEIVER, String.class);
+        if (receiver != null) {
```

Comment:
the test done before is '!Strings.isBlank(receiver)'  that checks if a String is whitespace, empty ("") or null. do you have a reasons for changing it ?
```suggestion
        !Strings.isBlank(receiver)
```

Reply:
Sometimes there is just a `!= null` test and sometimes `!Strings.isBlank(receiver)`. I tried to follow the same pattern when refactoring and sometimes kept both but indeed it doesn't make much sense. I will just use isBlank in both cases.